tests/tests/tier0/bluechi-change-port-with-c-cmd-option/test_bluechi_change_port_with_c_cmd_option.py

```diff
@@ -0,0 +1,61 @@
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+from typing import Dict
+import os
+
+from bluechi_test.test import BluechiTest
+from bluechi_test.container import BluechiControllerContainer, BluechiNodeContainer
+from bluechi_test.config import BluechiControllerConfig, BluechiNodeConfig
+from bluechi_test.util import read_file
+
+NODE_FOO = "node-foo"
+
+
+def exec(ctrl: BluechiControllerContainer, nodes: Dict[str, BluechiNodeContainer]):
+    node_foo = nodes[NODE_FOO]
+    config_file_location = "/var/tmp"
+    bluechi_agent_str = "bluechi-agent"
+    bluechi_controller_str = "bluechi-controller"
+    file_location_ctrl = os.path.join("config-files", "ctrl_port_8421.conf")
+    file_location_agent = os.path.join("config-files", "agent_port_8421.conf")
+
+    # Copying relevant config files into the nodes container
+    content = read_file(file_location_agent)
+    node_foo.create_file(config_file_location, file_location_agent, content)
+    content = read_file(file_location_ctrl)
+    ctrl.create_file(config_file_location, file_location_ctrl, content)
+
+    ctrl.restart_with_config_file(
+        os.path.join(config_file_location, "ctrl_port_8421.conf"), bluechi_controller_str)
+    assert ctrl.wait_for_unit_state_to_be(bluechi_controller_str, "active")
+
+    # Check if port 8421 is listenning and 8420 is disconnected
+    assert ctrl.exec_run("lsof -i:8421")[0] == 0
+    assert ctrl.exec_run("lsof -i:8420")[0] != 0
+
+    # Check if node disconnected
+    assert "Disconnected from manager" in node_foo.exec_run("systemctl status bluechi-agent.service")[1]
+
+    node_foo.restart_with_config_file(
+        os.path.join(config_file_location, "agent_port_8421.conf"), bluechi_agent_str)
+    assert node_foo.wait_for_unit_state_to_be(bluechi_agent_str, "active")
+    # Check if node disconnected
+    assert node_foo.exec_run("lsof -i:8421")[0] == 0
```

Comment:
What should this check do? Agent doesn't listen on port 8421, the agent is connecting to controller, which listens on port 8421. So whouldn't above be always true? Or am I missing something?

Reply:
As I see it we check if the controller listens on port 8421 in the code above. Then I check if the node successfully connected to the controller. The configuration for the agenda changes the port it should connect to the controller.
That is why I am checking if the controller listens on port 8421 and then if the agent is connected to that port.

Reply:
Hmm, so `lsof` displays also existing connections to a remote port and not only that port is open and some process listen on it locally? If so, then OK

Reply:
Correct